buildalias/buildalias.go

```diff
@@ -0,0 +1,201 @@
+	"2) allocate - create a new game server instance for the build alias and session id provided. Template for using\n\n\t" +
+	"buildalias allocate <alias-name> <sessionId>\n\n" +
+	"3) help - Displays this message and exit";
+
+	if len(args) == 1 {
+		fmt.Println(helpMessage);
+		return;
+	}
+
+	switch(args[1]) {
+
+	case "create":
+		CreateBuildAlias(args, fileName);
+
+	case "allocate":
+		AllocateForBuildAlias(args, fileName);
+
+	case "help":
+		fmt.Println(helpMessage);
+
+	default:
+		fmt.Println("Sorry, but the command "+ args[1] + " is not recognized.");
+	}
+}
+
+func CreateBuildAlias(args []string, fileName string) {
+	if len(args) < 5 {
+		log.Fatal("Not enough arguments were provided.");
+	}
+
+	fmt.Println("Creating the build alias: " + args[2]);
+
+	buildName := args[2];
+	aliasMap := map[string]map[string]int{};
+
+	aliasMap[buildName] = make(map[string]int);
+
+	for i := 3; i < len(args) - 1; i += 2 {
+		buildId := args[i];
+
+		if weight, err := strconv.Atoi(args[i+1]); err == nil {
+			aliasMap[buildName][buildId] = weight;
+		} else {
+			log.Fatal("Invalid weight " + args[i+1] + " provided for buildId: " + buildId + ". ", err);
+		}
+	}
+
+	yamlData, err := yaml.Marshal(&aliasMap);
+
+	if err != nil {
+		log.Fatal("Error while Marshaling. \n", err);
+	}
+
+	f, err := os.OpenFile(fileName, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644);
+
+	if err != nil {
+		log.Fatal("Error when creating the alias", err);
+	}
+
+	if _, err := f.Write(yamlData); err != nil {
+		f.Close();
+		log.Fatal(err);
+	}
+
+	if err := f.Close(); err != nil {
+		log.Fatal(err);
+	}
+
+	fmt.Println("Build alias " + buildName + " created successfully.");
+}
+
+func AllocateForBuildAlias(args []string, fileName string) {
+	if len(args) < 4 {
+		log.Fatal("Not enough arguments were provided.");
+	}
+
+	buildName := args[2];
+	sessionId := args[3];
+	var ipToAllocate string;
+
+	cmd := exec.Command("kubectl","get","svc","-n","thundernetes-system","thundernetes-controller-manager",
+			"-o","jsonpath='{.status.loadBalancer.ingress[0].ip}'");
+	commandOutput, err := cmd.CombinedOutput();
+
+	fmt.Println("Attempting to generate the IP address to allocate");
+	if err != nil {
+		log.Fatal("Error while trying to generate the IP, ", err);
+	}
+
+	if len(commandOutput) < 3 { // basically if we don't have a valid IP
+		// We will allocate locally
+		fmt.Println("Not able to generate the IP to allocate. Will use local environment");
+		ipToAllocate = "http://127.0.0.1:5000/api/v1/allocate";
+	} else {
+		ipToAllocate = string(commandOutput);
+		fmt.Println("IP", ipToAllocate, "to allocate generated successfully");
+
+		// We need to remove the '' at the begining and end of the IP address
+		ipToAllocate = strings.Trim(ipToAllocate, "'");
+		ipToAllocate = "http://" + ipToAllocate + ":5000/api/v1/allocate";
+	}
+
+	fmt.Println("Allocating a Game Server for " + buildName);

```

Comment:
the YAML file should be a Kubernetes ConfigMap and read using Kubernetes API

Reply:
can you please point me to the repo / library for this?

Reply:
https://kubernetes.io/docs/concepts/configuration/configmap/

We can either grab the ConfigMap via K8s API (https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/) or use something like this https://github.com/PlayFab/thundernetes/blob/main/pkg/operator/main.go#L229 (this is fetching a Secret instead of a ConfigMap but it should be similar). This is a very common practice in Kubernetes, picking up configuration from a ConfigMap. Thanks!